### PR TITLE
Pass in rest of args to _search_helper

### PR DESCRIPTION
**Pull request by OlajideOgun**
As of right now when trying to use functions like `max_marginal_relevance_search` or `max_marginal_relevance_search_by_vector` the rest of the `kwargs` are not propagated to `self._search_helper()`. For example a user cannot explicitly state the `distance_metric` they want to use when calling `max_marginal_relevance_search`

### CODE REVIEW

File: langchain/vectorstores/deeplake.py [hunk 1]
```diff
@@ -437,6 +437,7 @@ def max_marginal_relevance_search_by_vector(
             fetch_k=fetch_k,
             use_maximal_marginal_relevance=True,
             lambda_mult=lambda_mult,
+            **kwargs
```
Comment (skcoirz) [inline langchain/vectorstores/deeplake.py:440]: will be good to add annotation `Dict[str, Any]` and a trailing comma.

Comment (OlajideOgun) [reply langchain/vectorstores/deeplake.py:440]: Would it be better  I replace all the `Any` with `Dict[str, Any]`annotations for for `kwargs`

Comment (skcoirz) [reply langchain/vectorstores/deeplake.py:440]: theoretically yes, but when we touch more files, we will need to fully test all of them, which will increase the complexity of the commit. From my experience, I would usually suggest touching a few of them with solid test each time. Finding a good balance between code clean up efficiency and code change readability for reviewers. :)

Comment (dev2049) [reply langchain/vectorstores/deeplake.py:440]: yea lets type as Any for now

**Comment by OlajideOgun** [comment]
will close this and update a new

**Pull request closed without merging**